src/vm_pmu.rs

```diff
@@ -365,14 +362,14 @@ impl VmPmuState {
         }
     }
 
-    /// Calls the SBI configure_matching_counters() and performs internal bookkeeping on counter state.
-    pub fn configure_matching_counters(
+    fn configure_matching_counters_internal(
         &mut self,
         counter_index: u64,
         counter_mask: u64,
         config_flags: PmuCounterConfigFlags,
         event_type: PmuEventType,
         event_data: u64,
+        is_context_switch: bool,
```

Comment:
All this argument is used for is to set another flag in `config_flags`. How about doing that at the callsite instead?

Reply:
The issue happened (after refactoring) because the call-site set `PmuCounterStopFlags::default().set_reset_flag()`, and there was no way to tell it apart from the flags passed in by the VM. Note that it wasn't an issue before because the bookkeeping code wasn't part of the same function, so the extra flags were visible only to the SBI layer. However, it required the caller to make follow-up calls for bookkeeping and so forth, and having the callee handle everything is a better approach.

Reply:
Separated the bookkeeping code from the context switch path per offline discussion.